### PR TITLE
Added hrp to gateway interaction related code for hip data transfer

### PR DESCRIPTION
**Pull request by VenuAjitesh**
Request POJO's, Response POJO's, DataTransferInterface, and DataTransferService skeleton have been implemented.

### CODE REVIEW

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hip.hrp.dataTransfer;
+
+import com.nha.abdm.wrapper.common.responses.GatewayCallbackResponse;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.BundleResponseHIP;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.DataNotifyResponse;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.DataRequestResponse;
+
+public interface DataTransferInterface {
+  GatewayCallbackResponse DataOnNotifyCall(DataNotifyResponse data);
```
Comment (chandrashekar-s) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:10]: nit: Please follow `camelCasing` for method names, it should be `dataOnNotifyCall`.

Comment (atulai-sg) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:10]: also lets not use 'Call' in our method names

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hip.hrp.dataTransfer;
+
+import com.nha.abdm.wrapper.common.responses.GatewayCallbackResponse;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.BundleResponseHIP;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.DataNotifyResponse;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.responses.DataRequestResponse;
+
+public interface DataTransferInterface {
```
Comment (chandrashekar-s) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:9]: Please add documentation for the class and methods for easier understanding

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java (hunk 1 above)
Comment (chandrashekar-s) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:10]: Lets try to reuse the Pojo names similar to what ABDM has defined in their swagger.

**Review by chandrashekar-s — COMMENTED** [review]
Thanks @VenuAjitesh for the changes, there are few comments on improving the documentation. Please have a look.

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java (hunk 1 above)
Comment (VenuAjitesh) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:10]: Modified the naming conventions

Comment (VenuAjitesh) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:10]: Made the POJO class names similar with ABDM schema

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java (hunk 2 above)
Comment (VenuAjitesh) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferInterface.java:9]: Added documentation for the methods explaining the logic.

**Review by chandrashekar-s — APPROVED** [review]
@VenuAjitesh Thanks for the changes.

**Pull request merged**